common/logger/src/struct_log.rs

```diff
@@ -426,25 +421,34 @@ impl StructLogSink for UDPStructLog {
     }
 }
 
-struct UDPStructLogThread {
+struct TCPStructLogThread {
     receiver: Receiver<StructuredLogEntry>,
-    socket: UdpSocket,
-    udp_address: String,
+    address: String,
 }
 
-impl UDPStructLogThread {
+impl TCPStructLogThread {
     pub fn run(self) {
         for entry in self.receiver {
+            // Parse string
             let json_string = match entry.to_json_string() {
                 Ok(json_string) => json_string,
                 Err(_) => continue,
             };
-            if let Err(e) = self
-                .socket
-                .send_to(json_string.as_bytes(), self.udp_address.clone())
-            {
-                // do not break on error, move on to the next log message
-                println!("[Logging] Error while sending data to socket: {}", e);
+
+            // TODO: we should be able to share TcpStreams over messages
+            let mut stream = match TcpStream::connect(&self.address) {
```

Comment:
A few things:

* Add a connection timeout
* Add a write timeout
* Implement connection pooling / reuse -- my experience with vault suggests that the OSes don't like churning through sockets...

Reply:
They're all coming next!